sdk/core/src/services/anchoringHandler.ts

```diff
@@ -0,0 +1,65 @@
+import { ELEM_DID_METHOD, JOLO_DID_METHOD } from '../_defaultConfig'
+import { RegistryApiService } from '@affinidi/internal-api-clients'
+import { KeysService } from '@affinidi/common'
+
+export const anchorDid = async (
+  api: RegistryApiService,
+  encryptedSeed: string,
+  password: string,
+  didDocument: any,
+  anchoredDidElem: boolean = false,
+  nonce: number = 0,
+): Promise<{ did: string } | undefined> => {
```

Comment:
This function looks inconsistent to me.
1. Is it expected that we return only in case  `ELEM_DID_METHOD` and `anchoredDidElem`? We could return did in any case and then return type will be `Promise<{ did: string }>`.
2. We could initialise `anchorDidParams` and pass then where it is needed to avoid duplication:
```
const anchorDidParams = {
      did,
      didDocumentAddress: '',
      ethereumPublicKeyHex: '',
      transactionSignatureJson: '',
      anchoredDidElem,
 }
...
await api.anchorDid(anchorDidParams)
...
await api.anchorDid(anchorDidParams)
```
3. Why 3rd call to `api.anchorDid` wrapped by try-catch and first 2 not?